### PR TITLE
cmake: fancy colors for modern compilers

### PR DESCRIPTION
**Pull request by janisozaur**
a naive attempt at enabling colors in supporting compilers.

### CODE REVIEW

File: buildsystem/cpp.cmake [hunk 1]
```diff
@@ -12,26 +12,35 @@ function(cpp_init)
...
-
 	function(require_cxx_version CXXNAME MINIMAL)
 		if(CMAKE_CXX_COMPILER_VERSION VERSION_LESS ${MINIMAL})
 			message(FATAL_ERROR ">=${CXXNAME}-${MINIMAL} required (c++11, you know?), you have ${CMAKE_CXX_COMPILER_VERSION}")
 		endif()
 	endfunction()
 
+	function(set_cxx_versioned_flags CXXNAME MINIMAL FLAGS)
+		if(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS ${MINIMAL})
+			set(CMAKE_CXX_FLAGS "${FLAGS} ${CMAKE_CXX_FLAGS} ${FLAGS}" PARENT_SCOPE)
```
Comment (TheJJ) [inline buildsystem/cpp.cmake]: why is `flags` added two times?

Comment (janisozaur) [inline buildsystem/cpp.cmake]: Sorry about that, I didn't remove one while checking something

File: buildsystem/cpp.cmake [hunk 2]
```diff
@@ -12,26 +12,35 @@ function(cpp_init)
...
+	function(set_cxx_versioned_flags CXXNAME MINIMAL FLAGS)
+		if(NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS ${MINIMAL})
+			set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${FLAGS}" PARENT_SCOPE)
+		endif()
+	endfunction()
+
 	# check for compiler versions
 	if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
 		require_cxx_version("gcc" 4.8)
+		set_cxx_versioned_flags("gcc" 4.9 "-fdiagnostics-color")
```
Comment (TheJJ) [inline buildsystem/cpp.cmake]: Does it have to be `-fdiagnostics-color=always`? Probably this is `=auto` which might be better if one pipes the output to somewhere.

Comment (janisozaur) [inline buildsystem/cpp.cmake]: Yes, it is ```=auto``` by default

File: buildsystem/cpp.cmake [hunk 3]
```diff
@@ -12,26 +12,35 @@ function(cpp_init)
...
 	elseif("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
 		require_cxx_version("clang" 3.3)
+		set_cxx_versioned_flags("clang" 3.3 "-fdiagnostics-color")
```
Comment (TheJJ) [inline buildsystem/cpp.cmake]: Hm, my clang 3.5 manpage does not have a `-fdiagnostics-color` listed. Did you mean `-fcolor-diagnostics`?

**Comment by TheJJ** [comment]
lookin good now, merge: :+1:.

**Comment by fr3akout** [comment]
:+1:

**Pull request merged**